spring/boot2-autoconfigure/src/main/java/com/linecorp/armeria/internal/spring/ArmeriaServerGracefulShutdownLifecycle.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.spring;
+
+import org.springframework.context.SmartLifecycle;
+
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.spring.ArmeriaAutoConfiguration;
+
+/**
+ * Make Armeria {@link Server} utilize spring's SmartLifecycle feature.
+ * So Armeria will shutdown before other web servers and beans in the context.
+ */
+public final class ArmeriaServerGracefulShutdownLifecycle implements SmartLifecycle {
```

Comment:
I realized that we could just move this class to `com.linecorp.armeria.spring` package and make it package-private. 😅  Sorry about it.